### PR TITLE
Reverse a string in CPP

### PR DESCRIPTION
**Pull request by yashmita**
I have added the code for reversing a string using CPP for issue #611 . 
Kindly review it.

### CODE REVIEW

File: Code/C++/reverse_string.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*
+PROBLEM STATEMENT:
+Given a string, print its reverse.
```
Comment (raksha009) [inline Code/C++/reverse_string.cpp:3]: Plz explain a bit more by taking an example and then describing the workflow.

File: Code/C++/reverse_string.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+//Driver function
+int main(){
+	string str;
+	getline(cin, str);	//Taking input using getline() to include whitespaces
+	reverse(str.begin(), str.end());	//In-built function to reverse a string
```
Comment (raksha009) [inline Code/C++/reverse_string.cpp:13]: Write the code without include stl,  you may tell about this stl in the description or the comments in between the code.

File: Code/C++/reverse_string.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/*
+PROBLEM STATEMENT:
+Given a string, print its reverse.
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+//Driver function
```
Comment (raksha009) [inline Code/C++/reverse_string.cpp:9]: Remove this comment.

File: Code/C++/reverse_string.cpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+Given a string, print its reverse.
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+//Driver function
+int main(){
+	string str;
+	getline(cin, str);	//Taking input using getline() to include whitespaces
```
Comment (raksha009) [inline Code/C++/reverse_string.cpp:12]: Add comments above the line

File: string/readme.md [hunk 5]
```diff
@@ -13,3 +13,4 @@ A string is a class that contains a char array, but automatically manages it for
 
 ## Questions :
 * Revere a string using Stack ---->[C++](/Code/C++/reverse_a_string_using_stack.cpp)
+* Reverse a strung ---->[C++](/Code/C++/reverse_string.cpp)
```
Comment (raksha009) [inline string/readme.md:16]:
```suggestion
* Reverse a string ---->[C++](/Code/C++/reverse_string.cpp)
```

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: string/readme.md (hunk 5 above)
Comment (raksha009) [inline string/readme.md:16]:
```suggestion
* Reverse a string ---->[C++](/Code/C++/reverse_string.cpp)
```

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Review by raksha009 — APPROVED** [review]
LGTM 👍🏼

**Comment by einhawking** [comment]
I am a participant GSSOC'21
I would like to contribute in this

File: Code/C++/reverse_string.cpp [hunk 6]
```diff
@@ -0,0 +1,45 @@
+/*
+PROBLEM STATEMENT:
+Given a string, The task is to print its reverse. The approach is to traverse half of the string and swapping each character with a subsequent character present 
+in the other half of the string. For example, if the given string is "computer" then, we'll traverse the string till 'p' and swapping will be performed in the following way:
+c with r
+o with e
+m with t
+p with u
+At the end, we'll get "retupmoc", which is the required answer.
+We can also reverse a string using a predefined function in CPP i.e., reverse()
```
Comment (OjusWiZard) [inline Code/C++/reverse_string.cpp:10]: ...predefined function in **STL of** CPP i.e., **std::** reverse()

File: Code/C++/reverse_string.cpp [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	string str;
+	//Taking input using getline() to include whitespaces
+	getline(cin, str);	
+	//size() is used to find the length of the string
+	int len = str.size();
+	//Traversing half of the string
+	for(int i = 0; i < len/2; i++){
+		//swapping 
+		swap(str[i], str[len - 1 - i]);
+	}
```
Comment (OjusWiZard) [inline Code/C++/reverse_string.cpp:27]: Try adding the core algorithm in a separate function.

**Review by OjusWiZard — CHANGES_REQUESTED** [review]

**Comment by yashmita** [comment]
@OjusWiZard I have updated the changes as suggested. Kindly have a look.

**Review by OjusWiZard — APPROVED** [review]

**Pull request merged**